azurerm/config.go

```diff
@@ -189,7 +191,15 @@ func withRequestLogging() autorest.SendDecorator {
 
 func setUserAgent(client *autorest.Client) {
 	version := terraform.VersionString()
+	azureAgent := os.Getenv("AZURE_HTTP_USER_AGENT")
```

Comment:
out of interest - what's the use-case for this / where's this going to be set?

Reply:
The azure compute cloud shell sets this in their dockerfile (i.e. ENV AZURE_HTTP_USER_AGENT cloud-shell/1.0). This way we can collect some BI data on the provider/environment usage.